pdfminer/encodingdb.py

```diff
@@ -1,28 +1,56 @@
 
 import re
-from .psparser import PSLiteral
+
+import six  # Python 2+3 compatibility
+
 from .glyphlist import glyphname2unicode
 from .latin_enc import ENCODING
+from .psparser import PSLiteral
 
-import six # Python 2+3 compatibility
-
-STRIP_NAME = re.compile(r'[0-9]+')
+HEXADECIMAL = re.compile(r'[0-9a-fA-F]+')
 
 
-##  name2unicode
-##
 def name2unicode(name):
-    """Converts Adobe glyph names to Unicode numbers."""
-    if name in glyphname2unicode:
-        return glyphname2unicode[name]
-    m = STRIP_NAME.search(name)
-    if not m:
-        raise KeyError(name)
-    return six.unichr(int(m.group(0)))
+    """Converts Adobe glyph names to Unicode numbers.
+
+    In contrast to the specification, this raises a KeyError instead of return an empty string when the key is unknown.
+    This way the caller must explicitly define what to do when there is not a match.
+
+    Reference: https://github.com/adobe-type-tools/agl-specification#2-the-mapping
+
+    :returns unicode character if name resembles something, otherwise a KeyError
+    """
+    full_stop = u'\u002E'
```

Comment:
Can we move this to be a global variable instead of defining it inside the function? ( Like `HEXADECIMAL` ).